app.js

```diff
@@ -2,22 +2,120 @@
   if (!window.addEventListener) return // Check for IE9+
 
   let options = INSTALL_OPTIONS
+  let updateTimeout
+  const delay = 1500
+  const elements = []
+  const PAYPAL_SCRIPT_URL = "https://cdn.rawgit.com/paypal/JavaScriptButtons/master/dist/button.js"
+  const TIME_PERIOD_SYMBOLS = {
+    D: "Daily",
+    W: "Weekly",
+    M: "Monthly",
+    Y: "Yearly"
+  }
+
+  function updateElements() {
+    if (options.merchant) {
+      const {buttons} = options
+
+      buttons
+        .filter($ => $.name && $.amount)
+        .forEach((attrs, i) => {
+          const paypalButton = document.createElement("script")
+          const paypalInfoWrapper = document.createElement("eager-paypal-info-wrapper")
+          const paypalItemName = document.createElement("eager-paypal-item-name")
+          const paypalPrice = document.createElement("eager-paypal-price")
+
+          let currencySymbol
+          let times
+
+          if (options.region.currency === "GBP") {
+            currencySymbol = "£"
+          }
+          else {
+            currencySymbol = "$"
+          }
+
+          paypalItemName.innerHTML = `${attrs.name}`
+
+          paypalPrice.innerHTML = `${currencySymbol}${attrs.amount} ${options.region.currency} `
 
-  function updateElement() {
+          if (attrs.type === "subscribe") {
+            if (attrs.recurrence === 1){
+              times = "time"
+            }
+            else {
+              times = "times"
+            }
+            paypalPrice.innerHTML += `${attrs.recurrence} ${times} ${TIME_PERIOD_SYMBOLS[attrs.timePeriod]} `
+          }
+
+          if (options.region.tax && !attrs.shipping) {
```

Comment:
This is difficult to read because the order of the operators flips between the predicates (the expression in a statement).

Additionally much of the template code is common among all outcomes. Consider the following.



```javascript
// Above, pluck out region in a constant using destructuring like `buttons`

if (region.tax || attrs.shipping) {
  const additionalCost = region.tax + attrs.shipping

  let label

  if (region.tax && attrs.shipping) label = "shipping & tax"
  else if (region.tax) label = "tax"
  else if (attrs.shipping) label = "shipping"


  paypalPrice.innerHTML += `<small>+ ${currencySymbol}${additionalCost} ${label}</small>`
}
```